pkg/controller/gcp/cache/claim.go

```diff
@@ -73,26 +72,34 @@ func ConfigureCloudMemorystoreInstance(_ context.Context, cm resource.Claim, cs
 		return errors.Errorf("expected resource claim %s to be %s", cm.GetName(), cachev1alpha1.RedisClusterGroupVersionKind)
 	}
 
-	rs, csok := cs.(*corev1alpha1.ResourceClass)
+	rl, csok := cs.(*v1alpha1.CloudMemorystoreInstanceClass)
 	if !csok {
-		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), corev1alpha1.ResourceClassGroupVersionKind)
+		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), v1alpha1.CloudMemorystoreInstanceClassGroupVersionKind)
 	}
 
 	i, mgok := mg.(*v1alpha1.CloudMemorystoreInstance)
 	if !mgok {
 		return errors.Errorf("expected managed resource %s to be %s", mg.GetName(), v1alpha1.CloudMemorystoreInstanceGroupVersionKind)
 	}
 
-	spec := v1alpha1.NewCloudMemorystoreInstanceSpec(rs.Parameters)
+	// NOTE(hasheddan): Getting rid of the NewCloudMemorystoreInstanceSpec func creates a dependency on corev1alpha1.
```

Comment:
nit: i'm not sure this comment is worth persisting in the code base, i not thinking that it will be valuable to a someone else working on this code in the future.  since the `NewCloudMemorystoreInstanceSpec` function won't exist after this PR, it'll probably just confuse a future maintainer having to read and comprehend this comment.  consider removing it.

Reply:
Agreed! I think my initial thought process was that it will help if others are converting managed kinds to strongly typed in the future, but the body of this PR creates a pretty solid road map for that.